src/OrbitBase/OverloadedTest.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+#include <stdint.h>
+
+#include "OrbitBase/Overloaded.h"
+
+namespace orbit_base {
+
+constexpr std::string_view kInt = "int";
+constexpr std::string_view kString = "string";
+
+constexpr auto kFromInt = [](int /*unused*/) { return kInt; };
+constexpr auto kFromIntMutable = [](int /*unused*/) mutable { return kInt; };
+constexpr auto kFromString = [](std::string /*unused*/) { return kString; };
+
+std::string_view FromString(std::string /*unused*/) { return kString; }
+std::string_view FromInt(int /*unused*/) { return kInt; }
+
+TEST(OverloadedTest, TwoLambdas) {
```

Comment:
Can I request some more tests? :-)

- Multiple arguments
- Variadic lambdas (auto arguments)
- Stacked overloaded calls: `overloaded{overloaded{one, two}, three}`
- Function pointer to overload set. Does this work? Can this be made work?
- Some tests with move-only types is also usually a good idea because it tests whether your perfect forwarding setup works correctly.

Reply:
Thanks for the cool suggestions!

Reply:
@beckerhe sorry, forgot to ask.

> Function pointer to overload set. Does this work? Can this be made work?

How would this be possible without any wrappers like polymorphic lambda or `qOverload`? overloaded needs the types which cannot be deduced from `&Overloaded` alone.

Reply:
Yeah, I haven't thought about the details. But I guess it's not possible without a preprocessor macro that wraps the function pointer into some function object. From the top of my head I don't see a way to do this in generic C++.